### PR TITLE
PlatformManagerImpl improvements

### PR DESCRIPTION
**Pull request by ATmobica**
Set of improvements for Platform Manager and IoT socket implementation that should improve CI stability and performance.

### CODE REVIEW

File: src/platform/openiotsdk/OpenIoTSDKArchUtils.c [hunk 1]
```diff
@@ -46,31 +46,16 @@ uint64_t GetTick(void)
...
-
     return (uint32_t)(((uint64_t) ms * (uint64_t) osKernelGetTickFreq()) / 1000U);
 }
 
 uint32_t us2tick(uint32_t usec)
 {
-    if (usec == 0U)
-    {
-        return osWaitForever;
-    }
```
Comment (pan-) [inline src/platform/openiotsdk/OpenIoTSDKArchUtils.c:72]: Do you know why we were returning osWaitForever when the input was 0 ? 
The change looks correct with the usage made of the tick conversion functions.

File: src/platform/openiotsdk/PlatformManagerImpl.cpp [hunk 2]
```diff
@@ -152,74 +180,83 @@ void PlatformManagerImpl::HandleTimerEvent(void)
...
+        {
+            osEventFlagsSet(mPlatformFlags, kTaskHasEventLoopRunFlag);
         }
+    }
+
+    LockChipStack();
+
+    while (mRunEventLoop.load())
+    {
+        UnlockChipStack();
```
Comment (pan-) [inline src/platform/openiotsdk/PlatformManagerImpl.cpp]: Why are we Unlocking here ? Doesn't it means that the loop will be exited while the ChipStack is locked ? Why not unlock at the end of the loop ?

File: src/platform/openiotsdk/PlatformManagerImpl.cpp [hunk 3]
```diff
@@ -248,56 +278,53 @@ CHIP_ERROR PlatformManagerImpl::_StartEventLoopTask()
...
+#else
+    osEventFlagsSet(mPlatformFlags, kPostEventFlag);
 #endif // CHIP_SYSTEM_CONFIG_USE_IOT_SOCKET
 
-        osMutexRelease(mEventTaskMutex);
+    // If the thread running the event loop is different from the caller
+    // then wait it to finish
+    if (mEventTask != osThreadGetId())
+    {
+        if (osEventFlagsWait(mPlatformFlags, kTaskHasEventLoopStopFlag, osFlagsWaitAny, ms2tick(1000)) & osFlagsError)
```
Comment (pan-) [inline src/platform/openiotsdk/PlatformManagerImpl.cpp:328]: Why not use osThreadJoin ? 
Why do we have a timeout here ?

File: src/platform/openiotsdk/OpenIoTSDKArchUtils.c (hunk 1 above)
Comment (ATmobica) [reply src/platform/openiotsdk/OpenIoTSDKArchUtils.c:72]: I don't know but now it causes an issue in the event loop.

File: src/platform/openiotsdk/PlatformManagerImpl.cpp (hunk 2 above)
Comment (ATmobica) [reply src/platform/openiotsdk/PlatformManagerImpl.cpp]: The idea is to unlock the stack only for waiting for the event, Events service should be under the locked stack. We unlock when we leave the loop.

**Review by pan- — APPROVED** [review]

**Pull request merged**